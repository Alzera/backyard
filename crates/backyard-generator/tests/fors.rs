use backyard_generator::generate;
use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval("for ($i = 1; $i <= 10; $i++) {\n}").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn short() {
  let asts = parse_eval("for (;;):\nendfor;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn no_body() {
  let asts = parse_eval("for ($i = 1, $j = 0; $i <= 10; $j += $i, print $i, $i++);").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}
