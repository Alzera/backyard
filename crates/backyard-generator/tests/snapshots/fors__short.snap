---
source: crates/backyard-generator/tests/fors.rs
expression: generate(asts).unwrap()
snapshot_kind: text
---
"for (; ; ):\nendfor;"
